src/main/java/hudson/tasks/junit/TestResult.java

```diff
@@ -712,6 +798,16 @@ public void freeze(TestResultAction parent) {
             pr.freeze();
     }
 
+    private void addSuiteByRunAndNode(SuiteResult s) {
+        if (suitesByRunAndNode.get(s.getRunId()) == null) {
```

Comment:
Hmm. At least at first glance, I'm honestly not sure how to restructure this. For the moment, I'll add comments and if there's something you or anyone else sees as a better approach to the same end result, I'll happily embrace it.